Makefile

```diff
@@ -23,3 +23,35 @@ pod-lint:
 
 .PHONE: preflight-checks
 preflight-checks: pod-lint
+
+DOCS_DIR := docs
+DOCS_INDEX = $(DOCS_DIR)/index.html
+DOCS_README = readme.md
+DOCS_LICENSE = LICENSE.md
+LOWDOWN_PATH = $(shell which lowdown)
+MARKDOWN_PATH = $(shell which multimarkdown)
+
+ifneq ($(LOWDOWN_PATH),)
+	MARKDOWN_TOOL = $(LOWDOWN_PATH)
+	MARKDOWN_ARGS = "-so"
+else ifneq ($(MARKDOWN_PATH),)
+	MARKDOWN_TOOL = $(MARKDOWN_PATH)
+	MARKDOWN_ARGS = "-s"
+endif
+
+.PHONY: headerdoc
+headerdoc:
+	find MapboxMobileEvents -type f -name '*.h' | xargs headerdoc2html -o $(DOCS_DIR)
+	gatherheaderdoc $(DOCS_DIR)
+
+.PHONY: docindex
+docindex:
+	echo "<html><head><title>MapboxMobileEvents</title><meta http-equiv=\"content-type\" content=\"text/html; charset=utf-8\"><body>" > $(DOCS_INDEX)
+	$(MARKDOWN_TOOL) $(MARKDOWN_ARGS) $(DOCS_README) >> $(DOCS_INDEX)
+	echo "<h2>License</h2><tt>" >> $(DOCS_INDEX)
+	$(MARKDOWN_TOOL) $(MARKDOWN_ARGS) $(DOCS_LICENSE) >> $(DOCS_INDEX)
+	echo "</tt></body></html>" >> $(DOCS_INDEX)
+
+.PHONY: docs
+docs: $(DOCS_DIR) headerdoc docindex
```

Comment:
Consider using [jazzy](https://github.com/realm/jazzy/) instead of HeaderDoc. Unlike HeaderDoc, jazzy takes advantage of Clang’s built-in Doxygen parsing, so it doesn’t get tripped up by things like C attributes and macros, and it doesn’t require a separate Markdown parser.

[Here’s an example jazzy setup](https://github.com/mapbox/MapboxDirections.swift/blob/master/scripts/document.sh) and [configuration](https://github.com/mapbox/MapboxDirections.swift/tree/master/docs) in MapboxDirections.swift that applies a Mapbox-branded theme and customizes the cover page, though you might want to adopt [this shared Mapbox theme](https://github.com/mapbox/jazzy-theme/) at the same time. [Here’s what it looks like](http://docs.mapbox.com/ios/api/directions/).

/ref mapbox/mapbox-gl-native#3203
/cc @friedbunny @captainbarbosa